### PR TITLE
VxScan Diagnostic Scaffolding + Related Fixes

### PR DESCRIPTION
**Pull request by adghayes**
## Overview

Starts #1816. Closes #5178. Scaffolds out the VxScan Diagnostics and includes all of the pre-existing components:
- Configuration Info
- Storage Info
- Printer Status & Test Page Printing
- Saving as PDF

Required some refactor of the VxScan v4 printer user flows and a small change - instead of just printing and closing, the user is now prompted to confirm the print looks OK.

There needs to be a speaker diagnostic (#5193) and a scanner diagnostic (#5194) created still.

## Demo Video or Screenshot

https://github.com/user-attachments/assets/8f17179f-e51a-441b-8436-33ed9dfcbe58

## Testing Plan
- test coverage
- manual run through

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/scan/frontend/src/components/printer_management/election_manager_printer_tab_content.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ function StatusText({ printerStatus }: { printerStatus: PrinterStatus }) {
     case 'error':
       return (
         <P>
-          <Icons.X color="danger" />{' '}
+          <Icons.Danger color="danger" />{' '}
```
Comment (adghayes) [inline apps/scan/frontend/src/components/printer_management/election_manager_printer_tab_content.tsx:20]: Screen Shot 2024-08-01 at 3 02 07 PM

File: libs/ui/src/diagnostics/thermal_printer_section.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import { H2, P } from '../typography';
+import { InfoIcon, SuccessIcon, WarningIcon } from './icons';
+
+export interface ThermalPrinterSectionProps {
+  printerStatus: PrinterStatus;
+  mostRecentPrinterDiagnostic?: DiagnosticRecord;
+  printerActionChildren?: React.ReactNode;
+}
+
+function PrinterStatusDisplay({
```
Comment (adghayes) [inline libs/ui/src/diagnostics/thermal_printer_section.tsx:13]: Please sanity check this copy I came up with alongside Copilot

File: apps/scan/backend/src/util/diagnostics.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { assert, throwIllegalValue } from '@votingworks/basics';
+import { FujitsuPrinterStatus } from '../printing/printer';
+
+export function testPrintFailureDiagnosticMessage(
```
Comment (adghayes) [inline apps/scan/backend/src/util/diagnostics.ts:4]: Please sanity check this copy too.

File: apps/scan/frontend/src/components/printer_management/print_test_page_modal.tsx [hunk 4]
```diff
@@ -64,13 +87,34 @@ export function PrintTestPageModal({
     );
   }
 
+  if (
```
Comment (adghayes) [inline apps/scan/frontend/src/components/printer_management/print_test_page_modal.tsx:90]: Flow after modifications in this PR:

https://github.com/user-attachments/assets/d184e3e1-5b04-4547-8a96-9fbfa15fb970

**Comment by adghayes** [comment]
I decided to try removing the ballots scanned count from both system admin pages (root and diagnostics) and removing the election info bar from the diagnostics page, see commit message:

https://github.com/user-attachments/assets/1e347890-7373-48ea-b4aa-72eea80c7212

**Review by kofi-q — APPROVED** [review]
✅  LGTM

**Pull request merged**